.github/workflows/identify-new-migrations.yml

```diff
@@ -0,0 +1,186 @@
+      run: |
+        mysql -h 127.0.0.1 -uedxapp001 -ppassword -e "select 1;" edxapp
+        mongosh --host 127.0.0.1 --username edxapp --password password --eval 'use edxapp; db.adminCommand("ping");' edxapp
+
+    - name: Checkout master repo
+      uses: actions/checkout@v2
+      with:
+        ref: master
+
+    - name: Setup Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install system Packages
+      run: |
+        sudo apt-get update
+        make ubuntu-requirements
+
+    - name: Get pip cache dir
+      id: pip-cache-dir
+      run: |
+        echo "::set-output name=dir::$(pip cache dir)"
+
+    - name: Cache pip dependencies
+      id: cache-dependencies
+      uses: actions/cache@v3
+      with:
+        path: ${{ steps.pip-cache-dir.outputs.dir }}
+        key: ${{ runner.os }}-pip-${{ hashFiles('requirements/edx/development.txt') }}
+        restore-keys: ${{ runner.os }}-pip-
+
+    - name: Install Python dependencies
+      run: |
+        make dev-requirements
+        if [[ "${{ matrix.django-version }}" != "pinned" ]]; then
+          pip install "django~=${{ matrix.django-version }}.0"
+          pip check # fail if this test-reqs/Django combination is broken
+        fi
+
+    - name: Run migrations on master branch
+      env:
+        LMS_CFG: lms/envs/minimal.yml
+        # This is from the LMS dir on purpose since we don't need anything different for the CMS yet.
+        STUDIO_CFG: lms/envs/minimal.yml
+      run: |
+        echo "Running the LMS migrations."
+        ./manage.py lms migrate
+        echo "Running the CMS migrations."
+        ./manage.py cms migrate
+
+    - name: Verify executed migrations on master.
+      shell: bash
+      run: |
+        query_result1=$(mysql -h 127.0.0.1 -uedxapp001 -ppassword -e "select count(*) from django_migrations;" edxapp;)
+        echo "echo $query_result1"
+        numeric_result=$(echo "$query_result1" | tr -d '\n' | sed -E 's/[^0-9]+//g')
+        echo "Numeric Result: $numeric_result"
+        echo "FIRST_QUERY=$numeric_result" >> $GITHUB_ENV
+
+    - name: Checkout your branch repo
+      uses: actions/checkout@v2
+
+    - name: Install Python dependencies
+      run: |
+        make dev-requirements
+        if [[ "${{ matrix.django-version }}" != "pinned" ]]; then
+          pip install "django~=${{ matrix.django-version }}.0"
+          pip check # fail if this test-reqs/Django combination is broken
+        fi
+
+    - name: list installed package versions
+      run: |
+        pip freeze
+
+    - name: Run Tests
+      env:
+        LMS_CFG: lms/envs/minimal.yml
+        # This is from the LMS dir on purpose since we don't need anything different for the CMS yet.
+        STUDIO_CFG: lms/envs/minimal.yml
+      run: |
+        echo "Running the LMS migrations."
+        ./manage.py lms makemigrations
```

Comment:
Is this here to catch any code changes which should create migrations, but the author forgot to add them?

Reply:
Yes.